bridge_adaptivity/module/utils.py

```diff
@@ -59,4 +92,19 @@ def parse_callback_grade_xml(body):
     if not 0.0 <= score <= 1.0:
         raise BridgeError('score value outside the permitted range of 0.0-1.0')
 
-    return sourced_id, score
+    return imsx_message_identifier, sourced_id, score, action
+
+
+def chose_activity(sequence_item, sequence=None):
+    if sequence_item is None:
+        sequence = sequence
+    else:
```

Comment:
Do we really need increase complexity here?

Reply:
Actually, we have no choice and have to decide which sequence should be used, but I will rewrite this.